File: packages/sdk-snippets/src/fixtures/webhooks/quirky-escaping.js
module.exports = [
  {
    name: '"petstore" auth',
    default: 'default "key"\\',
    source: 'security',
    type: 'oauth2',
  },
  {
    name: 'basic-auth',
    default: 'default',
    source: 'security',
    type: 'http',
    scheme: 'basic',
  },
  {
    name: 'normal_security_var',
    default: 'default',
    source: 'security',
    type: 'http',
    scheme: 'basic',
  },
  {
    name: '2name',
    default: 'default-name',
    source: 'server',
  },
  {
    name: '*port',
    default: '',
    source: 'server',
  },
  {
    name: 'p*o?r*t',
    default: '',
    source: 'server',
  },
  {
    name: 'normal_server_var',
    default: '',
    source: 'server',
  },
];

File: packages/sdk-snippets/src/targets/csharp/dotnet6/webhooks/fixtures/quirky-escaping.cs
var builder = WebApplication.CreateBuilder(args);
var app = builder.Build();

// Your ReadMe secret
var secret = "my-readme-secret";

app.MapPost("/webhook", async context =>
{
  // Verify the request is legitimate and came from ReadMe.
  var signature = context.Request.Headers["readme-signature"];
  var body = await new StreamReader(context.Request.Body).ReadToEndAsync();

  try
  {
    ReadMe.Webhook.Verify(body, signature, secret);
  }
  catch (Exception e)
  {
    context.Response.StatusCode = StatusCodes.Status401Unauthorized;
    await context.Response.WriteAsJsonAsync(new
    {
      error = e.Message,
    });
    return;
  }

  // Fetch the user from the database and return their data for use with OpenAPI variables.
  // @todo Write your own query logic to fetch a user by `body["email"]`.

  await context.Response.WriteAsJsonAsync(new
  {
    // OAS Server variables
    "2name" = "default-name",
    "*port" = "",
    "p*o?r*t" = "",
    normal_server_var = "",

    // OAS Security variables
    "\"petstore\" auth" = "default \"key\"\\",
    "basic-auth" = new { user = "user", pass = "pass" },
    normal_security_var = new { user = "user", pass = "pass" },
  });
});

app.Run($"http://localhost:8000");
